### PR TITLE
Design patterns

### PR DESCRIPTION
**Pull request by Ivanenko16**

### CODE REVIEW

File: mavenProj.iml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ivanmatiashepam) [inline mavenProj.iml:1]: Please add this file to .gitignore and remove from git

File: src/main/java/desktop/pages/BookDepositoryLoginPage.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        typeText(emailField, str);
+        return this;
+    }
+
+    public BookDepositoryLoginPage setPassword(String str) {
+        typeText(passwordField, str);
+        return this;
+    }
+
+    public static String getStyleClearEmailButton() {
```
Comment (ivanmatiashepam) [inline src/main/java/desktop/pages/BookDepositoryLoginPage.java:60]: please make this method non-static. style of clear email button refers to some BookDepositoryLoginPage's object

File: src/test/java/runner/TestRunner.java [hunk 3]
```diff
@@ -1,11 +1,107 @@
...
 
 public class TestRunner {
-}
+
+    static WebDriver driver = SingletonDriver.getWebDriverInstance();
+
+    @Test
+    public void clearEmailButtonIsAppeared() {
+
+        new BookDepositoryLoginPage(driver)
```
Comment (ivanmatiashepam) [inline src/test/java/runner/TestRunner.java:19]: Please create a variable of BookDepositoryLoginPage inside each test and then getStyle for this variable (after you make that method non-static)

File: src/test/java/runner/TestRunner.java [hunk 4]
```diff
@@ -1,11 +1,107 @@
...
+                .openBookDepositoryLoginPage()
+                .switchToLogInFrame()
+                .setEmail("tractor@gmail.com");
+
+        String style = BookDepositoryLoginPage.getStyleClearEmailButton();
+
+        Assert.assertEquals("display: block;", style);
```
Comment (ivanmatiashepam) [inline src/test/java/runner/TestRunner.java:26]: Please add error message as a first argument to each assert

File: src/test/java/runner/TestRunner.java [hunk 5]
```diff
@@ -1,11 +1,107 @@
...
+        String url = driver.getCurrentUrl();
+
+        Assert.assertEquals(LOGIN_URL, url);
+    }
+
+
+    @AfterAll
+    static void browserTearDown() {
+        driver.quit();
+        driver = null;
```
Comment (ivanmatiashepam) [inline src/test/java/runner/TestRunner.java:105]: what does this line do?

File: src/main/java/driver/SingletonDriver.java [hunk 6]
```diff
@@ -1,5 +1,25 @@
...
 public class SingletonDriver {
 
+    private static WebDriver driver;
+
+    private SingletonDriver() {
+    }
+
+    public static WebDriver getWebDriverInstance() {
+        if (driver == null) {
+            System.setProperty(CHROME_DRIVER_EXE_PROPERTY, WEB_DRIVER_FILE_PATH);
```
Comment (ivanmatiashepam) [inline src/main/java/driver/SingletonDriver.java:18]: please move these lines to the constructor of SingletonDriver and call it from here (new SingletonDriver();)

**Review by ivanmatiashepam — CHANGES_REQUESTED** [review]

File: mavenProj.iml (hunk 1 above)
Comment (Ivanenko16) [reply mavenProj.iml:1]: added to the .gitignore

File: src/main/java/desktop/pages/BookDepositoryLoginPage.java (hunk 2 above)
Comment (Ivanenko16) [reply src/main/java/desktop/pages/BookDepositoryLoginPage.java:60]: fixed

File: src/test/java/runner/TestRunner.java (hunk 3 above)
Comment (Ivanenko16) [reply src/test/java/runner/TestRunner.java:19]: fixed

File: src/main/java/driver/SingletonDriver.java (hunk 6 above)
Comment (Ivanenko16) [reply src/main/java/driver/SingletonDriver.java:18]: fixed

File: src/test/java/runner/TestRunner.java (hunk 5 above)
Comment (Ivanenko16) [reply src/test/java/runner/TestRunner.java:105]: 105 row is delated

File: src/main/java/driver/SingletonDriver.java (hunk 6 above)
Comment (ivanmatiashepam) [reply src/main/java/driver/SingletonDriver.java:15]: please move next lines to the constructor as well. and in if-condition for getWebDriverInstance() method there should be only new SingletonDriver(); call

**Pull request merged**